src/test/java/com/contrastsecurity/maven/plugin/sdkx/ScanTest.java

```diff
@@ -1,25 +1,56 @@
 package com.contrastsecurity.maven.plugin.sdkx;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assumptions.assumeThat;
 
+import com.contrastsecurity.maven.plugin.Resources;
 import com.google.gson.Gson;
-import java.io.StringReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
 
 /** Unit tests for {@link Scan} */
 final class ScanTest {
 
-  @Test
-  void gson_deserialization_configuration() {
-    // GIVEN some JSON for a scan
-    final String id = "scan-id";
-    final String json = "{\"id\": \"" + id + "\"}";
+  private final Gson gson = GsonFactory.create();
 
+  @Test
+  void gson_deserialization_configuration() throws IOException {
     // WHEN deserialize with GSON
-    final Scan scan = new Gson().fromJson(new StringReader(json), Scan.class);
+    final Scan scan = readScanResource("/scan-api/scans/scan-completed.json");
 
     // THEN has expected ID
-    final Scan expected = new Scan(id);
+    final Scan expected = Scan.createCompleted("scan-id", "project-id", "organization-id");
     assertThat(scan).isEqualTo(expected);
   }
+
+  /** Verifies that completed scans and failed scans are finished */
+  @ValueSource(
+      strings = {"/scan-api/scans/scan-completed.json", "/scan-api/scans/scan-failed.json"})
+  @ParameterizedTest
+  void is_finished(final String path) throws IOException {
+    // TODO skip scan-failed.json because we do not yet have an example of a failed scan
```

Comment:
I tried like hell to get a scan to fail. The Scan team wasn't able to provide me with any examples. They showed me how to crash the scanner, and I still couldn't get one in the failed state. I'm not sure what to do here. I kept the TODO as a reminder to fill in this gap once we can get a scan to fail correctly.

What do you think we should do in the interim?

Reply:
This is addressed by extracting JSON to a file and reusing a utility for creating HTTP responses from JSON files

Reply:
Whoops, I didn’t realize you were doing `assumeThat` instead of `assertThat` for the `scan-failed` case. If your code handles the (potentially non-existent) "failed" case correctly, I’d lean towards enabling the "failed" test case and commenting that we’ve never seen a scan fail in the wild so we might remove it in the future. I think that’s clear enough and we can always remove later.

Reply:
`scan-failed.json` doesn't exist. If I enable that test case by removing the `assumeThat`, the tests will fail.

I can remove it altogether, but I like that it acts as a reminder that we need an example of a failed scan. Unfortunately no one can show me what that looks like (e.g. which fields are `null`).

What do you have in mind when you say "enable the failed test case"?

Reply:
I thought you had guessed at what a failed scan might look like and built `scan-failed.json`. Didn’t realize it doesn’t exist. What if we built scan-failed by copying the completed scan but changing only the `status` to `FAILED`? Seems like that would cover the case for now, and we could comment that it’s just a made up example and we’d like a real test case in the future.

Reply:
Also, should we add a test case for `CANCELLED` as well?

Reply:
Personally, I think guessing at what the scan-failed JSON looks like does more harm than good, because it builds false confidence, but I don't feel strongly enough to push back on this idea, so I'll make that change.

Yes, we should have a cancelled example, and I don't foresee any issues getting the API to give me an example of one.